### PR TITLE
podio: add version 1.1, edm4hep: add version 0.99.1

### PR DESCRIPTION
**Pull request by tmadlener**
Add the latest tags and update the c++ variant values for EDM4hep which is now c++20 only.

### CODE REVIEW

File: var/spack/repos/builtin/packages/edm4hep/package.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ class Edm4hep(CMakePackage):
     depends_on("nlohmann-json@3.10.5:")
     depends_on("podio@1:", when="@0.99:")
     depends_on("podio@0.15:", when="@:0.10.5")
-    for _std in _cxxstd_values:
+    for _std in ("17", "20"):
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/edm4hep/package.py]: I don't really like the repetition, but I guess we can live with it :-)

I think, you could define `_cxxstd_values = (conditional("17", when="@:0.99"), conditional("20"))` to have them both be same type (I'm assuming the conditional without a condition will work). Then you can loop
```suggestion
    for _cxxstd in _cxxstd_values:
        for _v in _cxxstd:
            depends_on(f"podio cxxstd={_v.value}", when=f"cxxstd={_v.value}")
```

**Review by wdconinc — DISMISSED** [review]
Checksums verified. Suggestion at your discretion.

File: var/spack/repos/builtin/packages/edm4hep/package.py (hunk 1 above)
Comment (tmadlener) [reply var/spack/repos/builtin/packages/edm4hep/package.py]: I wasn't aware that was even possible, so I didn't consider it. Done as suggested here now. **More importantly I also caught a version issue: it should be `conditional("17", when=@0.99.0)` to really work as expected**.

File: var/spack/repos/builtin/packages/edm4hep/package.py [hunk 2]
```diff
@@ -69,7 +70,9 @@ class Edm4hep(CMakePackage):
     depends_on("podio@1:", when="@0.99:")
     depends_on("podio@0.15:", when="@:0.10.5")
     for _std in _cxxstd_values:
-        depends_on("podio cxxstd=" + _std, when="cxxstd=" + _std)
+        assert len(_std) == 1
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/edm4hep/package.py]: I don't think the assert is needed here, right? Usually we avoid asserts in packages (and elsewhere for that matter) and handle the underlying failure modes when they can be identified. What's the failure mode here that you're worried about?

Comment (tmadlener) [reply var/spack/repos/builtin/packages/edm4hep/package.py]: Ah, this is a leftover from developing / debugging. Removed.

**Review by wdconinc — APPROVED** [review]
Thanks!

**Pull request merged**